### ISSUE TITLE
Add TUI for command `top`

### ISSUE THREAD

**Issue by Krysztal112233** [uutils/procps#191]
The current `top` command does not implement a TUI, it simply implements information collection and output.

**Comment by r3yc0n1c** [comment]
Hi, I’d like to work on this.

**Comment by cakebaker** [comment]
@r3yc0n1c sure, go ahead :)

**Comment by r3yc0n1c** [comment]
can we use something like https://ratatui.rs/ ?

**Comment by cakebaker** [comment]
Yes, I think that's fine. As I never used it before I don't know how suitable it is for this task, though.

**Comment by r3yc0n1c** [comment]
thanks... looking into it.

**Comment by r3yc0n1c** [comment]
hey @cakebaker, here's a POC of the TUI for `top`, built with the help of [ratatui.rs](https://ratatui.rs). Currently, the top block is static string as it still needs to be implemented in the `top.rs` main file. Everything else is dynamically pulled from the main `top.rs` file. The TUI exits when we press `'q'`.

**Comment by cakebaker** [comment]
Looks good :)
